### PR TITLE
much maintenance

### PR DESCRIPTION
**Pull request by johnlockwood-wf**
Years of maintenance fixes.

Fix rare hash collision issue.

Allow disabling of cpyamf coders.

Add support for appengine NDB.

Add feature to support casting attribute values to str/bytes for encoding.
Add feature to appengine db adapter to cast long property values to long for decoding.

Upgrade django tests to support django 1.4

### CODE REVIEW

File: pyamf/adapters/__init__.py [hunk 1]
```diff
@@ -34,11 +34,7 @@ def register_adapters():
     if adapters_registered is True:
         return
 
-    try:
-        import pkg_resources
-        packageDir = pkg_resources.resource_filename('pyamf', 'adapters')
-    except:
-        packageDir = os.path.dirname(__file__)
+    packageDir = os.path.dirname(__file__)
```
Comment (njoyce) [inline pyamf/adapters/__init__.py]: This is used if pyamf is in an egg.

File: pyamf/amf0.py [hunk 2]
```diff
@@ -234,13 +234,13 @@ def readMixedArray(self):
 
         attrs = self.readObjectAttributes(obj)
 
-        for key, value in attrs.iteritems():
+        for key in attrs.keys():
```
Comment (njoyce) [inline pyamf/amf0.py]: Why this change?

File: pyamf/amf3.py [hunk 3]
```diff
@@ -1093,13 +1097,14 @@ def readByteArray(self):
 
         buffer = self.stream.read(ref >> 1)
 
-        if buffer[0:2] == ByteArray._zlib_header:
```
Comment (njoyce) [inline pyamf/amf3.py]: This optimisation was here for a reason - why was it removed?

File: pyamf/amf0.py (hunk 2 above)
Comment (johnlockwood-wf) [inline pyamf/amf0.py]: looks like I chose wrong while merging this change from the WK version, which hadn't had https://github.com/hydralabs/pyamf/commit/dbe5d4e2aa7ff8527917ec0475bee7216108b56b applied.

File: pyamf/amf3.py (hunk 3 above)
Comment (johnlockwood-wf) [inline pyamf/amf3.py]: looks like I missed it from https://github.com/hydralabs/pyamf/commit/ce0440eba3220dc78d7643da56aedcc361e9f254 when merging the WK version

**Comment by njoyce** [comment]
All parts of this maintenance have now been added to master
